cloud.foundry.cli/src/main/java/cloud/foundry/cli/crosscutting/mapping/YamlMapper.java

```diff
@@ -41,19 +43,51 @@
      *
      * @param configFilePath the path to the config file
      * @param beanType the desired type of the bean to load
+     * @return a bean holding the information of the configuration file
      * @throws IOException if the config file cannot be accessed
      * @throws RefResolvingException if an error during the ref-resolution process occurs
-     * @throws ConstructorException if the resolved Object can not be dumped as the given Bean type
+     * @throws ParserException if the configuration file content cannot be parsed as yaml
+     * @throws ScannerException if the configuration file content cannot be interpreted as yaml
+     * @throws ConstructorException if the resolved config cannot be interpreted as the given bean type
      */
     public static <B extends Bean> B loadBean(String configFilePath, Class<B> beanType) throws IOException {
-        Object yamlTreeRoot = loadYamlTree(configFilePath);
+        String resolvedConfigContent = resolveYamlFile(configFilePath);
+        return interpretBean(resolvedConfigContent, beanType);
```

Comment:
Mind to explain this change? I think there should perhaps be a comment that summarizes what you're trying to achieve with this series of calls.

Reply:
I would argue that by the method names and the docstrings of this and the called methods, it is really well explained what is going on there